test/dispatch.jl

```diff
@@ -0,0 +1,91 @@
+@testset "type_morespecific" begin
+    @testset "subtype" begin
+        @test type_morespecific(Int, Integer)
+        @test !type_morespecific(Integer, Int)
+        @test !type_morespecific(Int, Int)
+    end
+
+    @testset "number of arguments" begin
+        @test type_morespecific(Tuple{Int,AbstractFloat}, Tuple{Integer})
+    end
+
+    @testset "ambiguous" begin
+        @test !type_morespecific(Tuple{Int,AbstractFloat}, Tuple{Integer,Float64})
+        @test !type_morespecific(Tuple{Integer,Float64}, Tuple{Int,AbstractFloat})
+    end
+
+    @testset "length" begin
+        @test type_morespecific(Tuple{Int,Int}, Tuple{Int,Vararg{Int}})
+    end
+end
+
+@testset "anonymous_signature" begin
+    plus_method = first(methods(+, (Int, Int)))
+    plus_sig = Tuple{typeof(+),T,T} where T <: Base.BitInteger
+    @test plus_method.sig == plus_sig
+
+    @test anonymous_signature(Tuple{typeof(+),Int,Int}) == Tuple{Int,Int}
+    @test anonymous_signature(plus_sig) == Tuple{T,T} where T <: Base.BitInteger
+    @test anonymous_signature(plus_method) == Tuple{T,T} where T <: Base.BitInteger
+end
+
+@testset "anon_morespecific" begin
+    plus_method = first(methods(+, (Int, Int)))
+    max_method = first(methods(max, (Int, Int)))
+
+    @test !type_morespecific(plus_method.sig, max_method.sig)
```

Comment:
Can a comment be added explaining what we are demonstrating here?

Reply:
Looking at this again I can make a simpler example. Mainly it's showing that the function type is dropped when performing the anonymous comparison

Reply:
Looking at this again I can make a simpler example. Mainly it's showing that the function type is dropped when performing the anonymous comparison

Reply:
Looking at this again I can make a simpler example. Mainly it's showing that the function type is dropped when performing the anonymous comparison